src/custom-element.js

```diff
@@ -352,14 +351,19 @@ function createBaseCustomElementClass(win) {
         // Already upgraded or in progress or failed.
         return;
       }
-      this.implementation_ = new newImplClass(this);
-      if (this.everAttached) {
-        // Usually, we do an implementation upgrade when the element is
-        // attached to the DOM. But, if it hadn't yet upgraded from
-        // ElementStub, we couldn't. Now that it's upgraded from a stub, go
-        // ahead and do the full upgrade.
-        this.tryUpgrade_();
+
+      // If the implementation fails to construct, we'll leave it as a
+      // ElementStub. The runtime will ignore the element from now on.
+      try {
+        this.implementation_ = new newImplClass(this);
+      } catch (e) {
+        this.upgradeState_ = UpgradeState.UPGRADE_FAILED;
+        const TAG = this.tagName;
+        dev().error(TAG, 'Failed to construct BaseElement', e);
```

Comment:
failed to construct `newImplClass.name` not BaseElement right?

Reply:
nevermind. all newImplClasses extend BaseElement

Reply:
Yup.